scripts/tests/api-docs.js

```diff
@@ -0,0 +1,177 @@
+
+const expect = chai.expect
+
+const pkg = process.argv[5].replace("--pkg=", "");
+
+console.log("pkg", pkg);
+
+if (!pkg) {
+  console.error("package not set", 1);
+}
+
+const modules = getModules(pkg);
+modules.forEach(mod => {
+  const paths = glob.sync(`./public/registry/packages/${pkg}/api-docs/${mod}/**/*.html`, {
+    ignore: [
+      `./public/registry/packages/${pkg}/api-docs/${mod}/index.html`
+    ],
+  });
+  paths.forEach(p => {
+    const fileContent = fs.readFileSync(p, "utf-8").toString();
+    const dom = htmlparser2.parseDocument(fileContent);
+    const $ = cheerio.load(dom);
+
+    // Ignore funtion pages for now.
+    if (!path.basename(path.dirname(p)).startsWith("get")) {
+
+      describe(constructPageRoute(p), function () {
+        // Verify page has a title and it is an h1 that contains the package and
+        // module name.
+        describe("h1 title", function () {
+          const h1s = $("h1");
+
+          it("contains exactly 1 h1", function () {
+            expect(h1s.length).to.equal(1);
+          })
+          it("contains the package name", function () {
+            expect(h1s.text().toLowerCase()).to.have.string(pkg);
+          })
+          it("contains the module name", function () {
+            expect(h1s.text().toLowerCase()).to.have.string(mod);
+          })
+        })
+
+        // Verify sections exist in correct order
+        describe("Sections", function () {
+          const headings = [];
+          $("section.docs-content > h2").each((i, elm) => {
+            if (elm) {
+              headings.push($(elm).text());
+            }
+          })
+          it('exist in correct order', function () {
+            const possibleHeadings = [
+              "Example Usage",
+              "Create",
+              "Resource Properties",
+              "Look up Existing",
+              "Supporting Types",
+              "Import",
+              "Package Details",
+            ];
+            const actual = possibleHeadings.filter(ph => {
+              return headings.some(h => {
+                return ph.includes(h);
+              })
+            });
+            const expected = possibleHeadings.filter(h => actual.includes(h))
+            expect(JSON.stringify(actual)).to.equal(JSON.stringify(expected));
+          })
+        })
+
+        // Verify it contains a description paragraph.
+        describe("Description paragraph", function () {
+          const paragraph = $(".docs-main-content .docs-content > p").text();
+          it('paragraph exists', function () {
+            expect(paragraph.length).to.be.at.least(1);
+          })
+        })
+
+        // Verify constructor section exists.
+        describe("Constructor", function () {
+          it("constructor section exists", function () {
+            const constructorSection = $("h2#create");
+            expect(constructorSection.length).to.equal(1);
+          })
+          it("syntax section exists", function () {
```

Comment:
Can we stick with arrow functions over anonymous functions?

```suggestion
          it("syntax section exists", () => {
```